File: src/main/java/com/prmncr/normativecontrol/factories/DocumentParserFactory.java
package com.prmncr.normativecontrol.factories;

import com.prmncr.normativecontrol.components.CorrectDocumentParams;
import com.prmncr.normativecontrol.components.SectorKeywords;
import com.prmncr.normativecontrol.services.DocumentParser;
import lombok.AllArgsConstructor;
import org.docx4j.openpackaging.packages.WordprocessingMLPackage;
import org.springframework.context.annotation.Configuration;

@Configuration
@AllArgsConstructor
public class DocumentParserFactory {
    private CorrectDocumentParams params;
    private SectorKeywords keywords;

    public DocumentParser build(WordprocessingMLPackage docx) {
        return new DocumentParser(docx, params, keywords);
    }
}

File: src/main/kotlin/com/maeasoftworks/normativecontrol/components/DocumentParserFactory.kt
package com.maeasoftworks.normativecontrol.components

import com.maeasoftworks.normativecontrol.dtos.Document
import com.maeasoftworks.normativecontrol.dtos.DocumentParser
import org.springframework.stereotype.Component

@Component
class DocumentParserFactory(
    private val params: CorrectDocumentParams,
    private val keywords: HeadersKeywords
) {
    fun create(document: Document): DocumentParser {
        return DocumentParser(document, keywords)
    }
}